Sources/XCLogParser/lexer/Scanner.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+
+final class Scanner {
+
+    let string: String
+
+    private(set) var offset: Int
+    private(set) lazy var stringEndIndex: String.Index = self.string.endIndex
+
+    var isAtEnd: Bool {
+        String.Index(compilerSafeOffset: self.offset, in: self.string) >= self.stringEndIndex
+    }
+
+    init(string: String) {
+        self.string = string
+        self.offset = 0
+    }
+
+    func scan(count: Int) -> String? {
+        let start = String.Index(compilerSafeOffset: self.offset, in: self.string)
+        let end = String.Index(compilerSafeOffset: self.offset + count, in: self.string)
+
+        let result = self.string[start..<end]
```

Comment:
Found a few index out of bounds crashes in this code, which is fixed by:

```suggestion
        let endOffset = self.offset + count

        guard endOffset <= self.string.utf16.count else { return nil }

        let end = String.Index(compilerSafeOffset: endOffset, in: self.string)
        let result = self.string[start..<end]
```

Reply:
@jrcmramos-bumble would you be able to look at the above?